ipysheet/test_all.py

```diff
@@ -54,7 +54,7 @@ def test_calculation():
     b = ipysheet.cell(0, 0, value=2)
     c = ipysheet.cell(0, 0, value=0)
 
-    @ipysheet.calculation(inputs=[a, b], output=c)
+    @ipysheet.calculation(inputs=[a, (b, 'value')], output=c)
```

Comment:
I'm not sure what you are trying to achieve in the test? Your cell value is an integer, it does not have a 'value' attribute

Reply:
The idea was that it's a list of [widget, (widget, 'traitname'), ...], and I wanted to test that last codepath. However, you also handle the case with the cell.value beging a widget differently (smart!). I added some test coverage for that, and applied some fixes (maybe border cases, insane!)